File: 74-search-a-2d-matrix/main.go
func searchMatrix(matrix [][]int, target int) bool {
    m, n := len(matrix), len(matrix[0])

    left, right := 0, m * n -1
    for left <= right {
        mid := left + (right - left) / 2
        i, j := mid / n, mid % n
        if target == matrix[i][j]{
            return true
        }else if target  > matrix[i][j]{
            left = mid + 1
        }else {
            right = mid -1
        }
    }

    return false
}

File: 74-search-a-2d-matrix/main.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m = len(matrix)
        n = len(matrix[0])


        left, right = 0, m * n - 1
        while left <= right:
            mid = left + (right - left) // 2
            i, j = mid // n, mid % n

            if target == matrix[i][j]:
                return True
            elif target > matrix[i][j]:
                left = mid + 1
            else:
                right = mid - 1

        return False
